docs/guide/advanced/navigation-failures.md

```diff
@@ -0,0 +1,57 @@
+# Navigation Failures
+
+When using `router-link`, Vue Router internally calls `router.push` to trigger a navigation. Depending on the current location and existing [Navigation Guards](./navigation-guards.md), this navigation might end up in a new page being shown, but there are a couple of situations where we will stay on the same page:
+
+- We are already on the page we are trying to go to
+- A navigation guard aborts the navigation by calling `next(false)`
+- A navigation guard throws an error or calls `next(new Error())`
+
+When using a regular `router-link`, **none of these failures will log an error**. However, if you are using `router.push` or `router.replace`, you might come across an _"Uncaught (in promise) Error"_ message followed by a more specific message in your console. This is part of the _Navigation Failures_ system in Vue Router and it is only available from version 3.2.0 onwards but existed for a long time before: through the two optional callbacks, `onComplete` and `onAbort` that can be passed to `router.push`. Since version 3.1.0, `router.push` and `router.replace` return a _Promise_ if no `onComplete`/`onAbort` callback is provided. This _Promise_ resolves instead of invoking `onComplete` and rejects instead of invoking `onAbort`.
+
+## Detecting Navigation Failures
+
+_Navigation Failures_ are `Error` instances with a few extra properties. To check if an error comes from the Router, use the `isRouterError` function:
+
+```js
+import { NavigationFailureType, isRouterError } from 'vue-router'
```

Comment:
Maybe something like `isNavigationFailure` is better naming than `isRouterError` as there aren't any other errors and it's also confusing for it to be named error and failure

Reply:
Agree. I would also lean towards `isRoutingError` given that this is an `Error` instance. So either `isNavigationError` or `isRoutingError`.

Reply:
Oh I was going the other way around, it's an Error instance to have a stack trace but it's technically not an error as it's expected behavior. Maybe Failure isn't a good name either

Reply:
What about `Rejection`?

Reply:
It’s much softer than both `Error` and `Failure` and might be quite good at communicating what is actually happening, as in this navigation request has been rejected. Not an error (this is the expected behaviour/feature). This would also allow us to actually have Errors thrown that are actual errors (as in unexpected).

Reply:
Yeah, it's softer but it's also related to Promise rejection and in v4, they appear when the promise resolve. It would also require changing the `NavigationRejectionType` name, but that could only be done in v4 and add an amendment to the RFC that was merged

Reply:
That makes sense! In that case maybe Exception? Running out of ideas here :(

Reply:
There is also NavigationDropped, Canceled (one of the types), Stopped

Reply:
In that case, I think Failure is a good word.